### PR TITLE
Add a disconnect button to the active connection notification

### PR DESCRIPTION
**Pull request by jklein24**

### CODE REVIEW

File: app/src/main/java/org/connectbot/HostListActivity.java [hunk 1]
```diff
@@ -394,13 +431,37 @@ public void onClick(DialogInterface dialog, int which) {
...
+		// Copy the bridges list because bridges are removed from the array when disconnected.
+		for (TerminalBridge bridge : new ArrayList<TerminalBridge>(bound.bridges)) {
+			bridge.dispatchDisconnect(true);
+		}
+		updateHandler.sendEmptyMessage(-1);
+		waitingForDisconnectAll = false;
+
+		if (closeOnDisconnectAll) {
+			// Clear the intent so that the activity can be relaunched without closing.
+			// TODO(jlklein): Find a better way to do this.
```
Comment (jklein24) [inline app/src/main/java/org/connectbot/HostListActivity.java]: @kruton, do you know of a better way to do this? The details of figuring out when this activity should actually be closed are really tricky and seem error-prone. Basically, I only want to close the activity if it was only launched to do the disconnect.

File: app/src/main/java/org/connectbot/HostListActivity.java [hunk 2]
```diff
@@ -129,11 +145,33 @@ public void onStop() {
...
 	}
 
 	@Override
 	public void onResume() {
 		super.onResume();
+
+		// Must disconnectAll before setting closeOnDisconnectAll to know whether to keep the
+		// activity open after disconnecting.
+		if ((getIntent().getFlags() & Intent.FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY) == 0 &&
+				getIntent().getAction() == DISCONNECT_ACTION) {
```
Comment (kruton) [inline app/src/main/java/org/connectbot/HostListActivity.java]: DISCONNECT_ACTION.equals(getIntent().getAction())

File: app/src/main/java/org/connectbot/HostListActivity.java (hunk 1 above)
Comment (kruton) [inline app/src/main/java/org/connectbot/HostListActivity.java]: This should probably go into `TerminalManager`. Also, if you receive the `DISCONNECT_ACTION`, you should probably always `finish()`. The user doesn't want to continue interacting with the app if they're closing everything, right?

File: app/src/main/java/org/connectbot/HostListActivity.java (hunk 2 above)
Comment (jklein24) [inline app/src/main/java/org/connectbot/HostListActivity.java]: Done.

File: app/src/main/java/org/connectbot/HostListActivity.java (hunk 1 above)
Comment (jklein24) [inline app/src/main/java/org/connectbot/HostListActivity.java]: Perhaps, but then should that also be the case if the user disconnects individually from all connections using the context menus? I feel like if the user was already in the hostlistactivity and then presses disconnect all, they should stay where they were. I could be convinced either way, though. 

sigh... totally missed that there's already a `disconnectAll` in TerminalManager. Updated.

**Pull request merged**